commands/Moderation/nickname.js

```diff
@@ -0,0 +1,55 @@
+exports.run = async (client, msg, [user, name]) => {
+  const { reportChannelId, titleURL } = msg.guild.settings;
+  try {
+    // checks to see if the nickname is too long or too short for the discord requirement if so replies and then exits.
+    if (name.length > 32 || name.length === 0) {
+      msg.reply("Sorry the nickname was too long or too short.");
+      return;
+    }
+    // set the nickname
+    msg.mentions.members.first().setNickname(`${name}`);
+    // send a message in regards to what changed in an embed.
+    const embed = new client.methods.Embed()
+      .setTitle("Nickname has been changed!")
+      .setAuthor(client.user.username, client.user.avatarURL)
+      // Alternatively, use "#00AE86", [0, 174, 134] or an integer number.
+      .setColor(0x00AE86)
+      .setDescription(`Nickname has been changed to ${name} for ${user} :smiley:`)
+      .setFooter("Made with Komada, Discord.js, & <3", client.user.avatarURL)
+      // Takes a Date object, defaults to current date.
+      .setTimestamp()
+      .setURL(titleURL);
+    msg.reply({ embed });
+  } catch (e) {
+    msg.reply("Some error occured with un-muting the member. A report has been sent to the developers.");
+    client.channels.get(reportChannelId).send(`There was an error trying to un-mute: ${e} in ${msg.channel} on ${msg.guild} by ${msg.author}`);
+  }
+};
+
+exports.conf = {
+  enabled: true,
+  runIn: ["text"],
+  aliases: ["nick"],
+  permLevel: 2,
+  botPerms: ["CHANGE_NICKNAME", "MANAGE_NICKNAMES"],
```

Comment:
When changing the bot's nickname, it doesn't require MANAGE_NICKNAMES, but only CHANGE_NICKNAME. The second one is to change somebody else's nickname.

Reply:
That's what the command is. It is for changing another users nickname. For example, I have people join the server with really filthy names and the mods can use this command to edit their nickname.

I was just thinking that if the user also wanted to change bots nickname they would need CHANGE perm also

Reply:
Cannot confirm, but I'm pretty sure MANAGE_NICKNAMES overrides CHANGE_NICKNAME

Reply:
Can confirm.